insights/src/main/kotlin/com/algolia/instantsearch/insights/database/DatabaseSharedPreferences.kt

```diff
@@ -0,0 +1,35 @@
+package com.algolia.instantsearch.insights.database
+
+import android.content.Context
+import com.algolia.instantsearch.insights.converter.ConverterEventToString
+import com.algolia.instantsearch.insights.converter.ConverterStringToEvent
+import com.algolia.instantsearch.insights.event.Event
+
+internal class DatabaseSharedPreferences(
+    private val context: Context,
+    override val indexName: String
+) : Database {
+
+    private val preferences = context.sharedPreferences(indexName)
```

Comment:
Would prefix `indexName` with something like `IS_INSIGHTS_`, as the risk of collision seems quite high (what if my app is called Aviato, I have an `aviato` Algolia index and use an `aviato` SharedPreferences key?)